### ISSUE TITLE
transformer lm training

### ISSUE THREAD

**Issue by JohnWick2ETH** [JohnWick2ETH/llm-learning-journey#8]
we need to implement the following components: 

- [x] AdamW optimizer and learning rate schedule
- [x] cross entropy loss function
- [x] checkpointing
